app/code/community/Algolia/Algoliasearch/Helper/IndexChecker.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+class Algolia_Algoliasearch_Helper_IndexChecker extends Mage_Core_Helper_Abstract
+{
+    /** @var Algolia_Algoliasearch_Helper_Config */
+    private $configHelper;
+    
+    /** @var Varien_Db_Adapter_Interface */
+    private $dbConnection;
+    
+    /** @var array */
+    private $pendingProductIds;
+
+    public function __construct()
+    {
+        /** @var Algolia_Algoliasearch_Helper_Config configHelper */
+        $this->configHelper = Mage::helper('algoliasearch/config');
+    }
+
+    /**
+     * @param $storeId
+     * @param $productIds
+     * @throws Exception
+     */
+    public function checkIndexers($storeId, $productIds)
+    {
+        if ($this->configHelper->isQueueActive($storeId) === false) {
+            return;
+        }
+        
+        if (method_exists('Mage', 'getEdition') === false) {
+            return;
+        }
+
+        if (Mage::getEdition() !== Mage::EDITION_ENTERPRISE) {
+            return;
+        }
+
+        if (!is_array($productIds)) {
+            $productIds = array($productIds);
+        }
+
+        if (empty($productIds)) {
+            return;
+        }
+
+        $pendingProductIds = $this->getPendingProductIds($storeId);
+
+        if (empty($pendingProductIds)) {
+            return;
+        }
+
+        foreach ($productIds as $id) {
+            if (isset($this->pendingProductIds[$id])) {
+                // Throw an exception - this exception is caught in Queue class and it'll retry next time Algolia is processed.
+                throw new Exception('Price index still pending for entity ID ' . $id);
```

Comment:
It is probably worth updating this exception message as you do not know if it is the stock or price index that is still pending. (If you wanted you could search for the price and stock index separately, however I joined them as the two arrays could have some overlaps.) A more generic message that I used in the logs was "Skipping, index still pending for entity id"